resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala

```diff
@@ -395,6 +395,14 @@ private[spark] object Config extends Logging {
         " positive time value.")
       .createWithDefaultString("30s")
 
+  val KUBERNETES_EXECUTOR_API_POLLING_WITH_RESOURCE_VERSION =
+    ConfigBuilder("spark.kubernetes.executor.enablePollingWithResourceVersion")
+      .doc("If true, `resourceVersion` is set with `0` during invoking pod listing APIs " +
+        "in order to allow API Server-side caching. This should be used carefully.")
```

Comment:
Now we replace only when we receive a monotonically increased resourceVersion. Is `This should be used carefully` still necessary?

Reply:
Yes, it's still risky because it gets stale data~

Reply:
Spark driver still may time-out executor pods because they are not visible to the driver.